src/app/sidenav/sidenav.component.html

```diff
@@ -3,9 +3,20 @@
     <mat-nav-list>
       <a mat-list-item [routerLink]="['/']">Settings</a>
       <mat-divider></mat-divider>
-      <a mat-list-item [routerLink]="['/']">Community 1</a>
-      <a mat-list-item [routerLink]="['/']">Community 2</a>
-      <a mat-list-item [routerLink]="['/']">Community 3</a>
+      <div *ngFor="let communityFav of communitiesFavorited; let j = index">
+        <a mat-list-item [routerLink]="['/']" *ngIf="communityFav.isFavorited">
+          <button mat-icon-button (click)="removeFavoriteFromFav(j)" ><mat-icon class="favorited">star</mat-icon></button>
+           {{ communityFav.communityname }}
+        </a>
+      </div>
```

Comment:
I've been playing around with these elements in order to fix your issue. The biggest thing that arose was that when the user decides to favourite a community, we also didn't want them to also trigger the link of the button (or reload the page in your case).

You can remove all of the special positioning styling (except for changing the favourite highlight colour) with the following structure.

```html
<mat-list-item *ngFor="let communityFav of communitiesFavorited; let j = index">
        <a mat-line *ngIf="communityFav.isFavorited" [routerLink]="['/']">
           {{ communityFav.communityname }}
        </a>
        <button mat-icon-button (click)="removeFavoriteFromFav(j)" ><mat-icon>star</mat-icon></button>
</mat-list-item>
```

It changes the main element to the mat-list-item and moves the button outside of the link. This is how they do it in the demo app, so I'd assume it'd have better scaling and all that.